### ISSUE TITLE
Always working GC in debug mode Xamarin Forms 5

### ISSUE THREAD

**Issue by Alex-Dobrynin** [xamarin/xamarin-android#5752]
Don't know to which team this bug is related, to VS or to Xamarin Android or even to XF, but in debug mode GC is not stopping and application becomes not responsible. When I unplug my phone from PC and restart the app it works perfectly. Don't know how to reproduce this. I beleive it is so rare bug. Only in debug mode. Can reproduce on 2 PC's and on different devices including emulators
2 weeks ago I didnt notice such behavior.

![image](https://user-images.githubusercontent.com/23138430/111368172-2fcf4680-869e-11eb-9eff-e0002cf6b261.png)

and this is infinite.

In iOS everything OK, but iOS I run on mac

Maybe when gather more info I will update this post

XF 5, VS 16.8.5, 16.9.1

#https://github.com/xamarin/Xamarin.Forms/issues/14020

**Comment by olenavasylchak** [comment]
Same problem(

**Comment by lambdageek** [comment]
This does not look like a GC problem.
From the log you can see that with each collection, objects are being collected and promoted. So looks like allocations keep happening forever from the framework side.

**Comment by DanyloKD** [comment]
This issue seems to be connected with Hot Reload. At least, I was able to fix continuous GC collections by disabling that feature.
Debug -> Options -> Hot Reload -> disable checkbox for: Android and iOS

I've also noticed that frequent calls to constructors of UI elements happening parallel to abovementioned GC collections. It supports what @lambdageek mentioned earlier - frequent allocations. This also correlates well with theory about issue in Hot Reload.

About a week ago, we've noticed warning in VS saying that it is not recommended to use Fast deployment and Linker simultaneously, and changed Linking setting to None in Debug configuration. Changing that setting back to "SDK Assemblies Only" (obviously) breaks Hot Reload, but prevents GC collections.
